grades/migrations/0014_remove_micromasterscoursecertificate_final_grade.py

```diff
@@ -20,33 +21,51 @@ def delete_duplicate_cert(apps, schema_editor):
             course=course
         ).values('user').annotate(Count('user')).filter(user__count__gt=1)
         for user in users_with_dup:
-            mmtrack = get_mmtrack(user['user'], course.program)
-            best_grade = mmtrack.get_best_final_grade_for_course(course)
+            best_grade = FinalGrade.objects.filter(
+                user=user['user'],
+                course_run__course=course,
+                passed=True
+            ).order_by('-grade').first()
             # trying to preserve the cert that is linked in the dashboard
             if MicromastersCourseCertificate.objects.filter(final_grade=best_grade).exists():
                 # delete other certificates
                 MicromastersCourseCertificate.objects.filter(
-                    course=course, user=user
+                    course=course, user=user['user']
                 ).exclude(final_grade=best_grade).delete()
             else:
                 certs = MicromastersCourseCertificate.objects.filter(
-                    course=course, user=user
+                    course=course, user=user['user']
                 )
                 certs.exclude(id__in=certs[:1]).delete()
 
 
 def add_final_grade(apps, schema_editor):
     MicromastersCourseCertificate = apps.get_model('grades', 'MicromastersCourseCertificate')
     FinalGrade = apps.get_model('grades', 'FinalGrade')
+    Course = apps.get_model('courses', 'Course')
 
+    certificates= MicromastersCourseCertificate.objects.filter(course__isnull=True)
+    for certificate in certificates:
+        certificate.delete()
     certificates = MicromastersCourseCertificate.objects.filter(course__program__financial_aid_availability=True)
+
     for certificate in certificates:
-        certificate.final_grade = FinalGrade.objects.filter(
+        passing_final_grades = FinalGrade.objects.filter(
             user=certificate.user,
             course_run__course=certificate.course,
             passed=True
-        ).order_by('-grade').first()
+        ).order_by('-grade')
+        certificate.final_grade = passing_final_grades.first()
         certificate.save()
+        # check if has other passing final_grades
+        for final_grade in passing_final_grades[1:]:
+            hash = generate_md5('{}|{}'.format(certificate.user_id, final_grade.course_run_id).encode('utf-8'))
+            MicromastersCourseCertificate.objects.create(
+                final_grade=final_grade,
```

Comment:
final grade is not part of MicromastersCourseCertificate

Reply:
this is a reverse migration to add it back